src/topology/homotopy/basic.lean

```diff
@@ -57,52 +57,68 @@ noncomputable theory
 
 universes u v w
 
-variables {X : Type u} {Y : Type v} {Z : Type w}
+variables {F : Type*} {X : Type u} {Y : Type v} {Z : Type w}
 variables [topological_space X] [topological_space Y] [topological_space Z]
 
 open_locale unit_interval
 
 namespace continuous_map
 
-/--
-The type of homotopies between two functions.
--/
+/-- `continuous_map.homotopy f₀ f₁` is the type of homotopies from `f₀` to `f₁`.
+
+When possible, instead of parametrizing results over `(f : homotopy f₀ f₁)`,
+you should parametrize over `(F : Type*) [homotopy_class F f₀ f₁] (f : F)`.
+
+When you extend this structure, make sure to extend `continuous_map.homotopy_class`. -/
 structure homotopy (f₀ f₁ : C(X, Y)) extends C(I × X, Y) :=
-(to_fun_zero : ∀ x, to_fun (0, x) = f₀ x)
-(to_fun_one : ∀ x, to_fun (1, x) = f₁ x)
+(map_zero_left' : ∀ x, to_fun (0, x) = f₀ x)
+(map_one_left' : ∀ x, to_fun (1, x) = f₁ x)
+
+/-- `continuous_map.homotopy_class F f₀ f₁` states that `F` is a type of homotopies between `f₀` and
+`f₁`.
+
+You should extend this class when you extend `continuous_map.homotopy`. -/
+class homotopy_class (F : Type*) (f₀ f₁ : out_param $ C(X, Y))
```

Comment:
I'm not so convinced by this name, since "homotopy class" is already a mathematical concept.